region/src/main/java/org/motechproject/nms/region/service/impl/DistrictServiceImpl.java

```diff
@@ -107,13 +124,23 @@ public District execute(Query query) {
     }
 
     @Override
-    public void create(District district) {
-        districtDataService.create(district);
+    @CacheEvict(value = {"district-state-name", "district-state-code", "district-language" }, allEntries = true)
+    public District create(District district) {
```

Comment:
Is there a reason eviction is done as an annotation on create/update methods as opposed to a lifecycle event that was used elsewhere?